CIP-0035/README.md

```diff
@@ -0,0 +1,302 @@
+[^3]: See "Why do performance changes require extra steps?".
+[^4]: See "Why do performance changes require extra steps?".
+
+## Specification
+
+### Changes that require a CIP
+
+Changes that require a CIP do not have to each be in an individual CIP, they can be included in batches or in other CIPs. 
+So, for example, a single CIP could propose multiple new builtin functions, or a CIP proposing a change to the ledger could also propose a change to the ledger-script interface.
+
+The requirement to propose a change via a CIP is, as all CIPs are, advisory. 
+In exceptional circumstances or where swift action is required, we expect that changes may still be made without following this process. 
+In such circumstances, a retrospective CIP SHOULD be made after the fact to record the changes and the rationale for them.
+
+#### Additions to the Plutus Core Builtins
+
+Proposals for additions to the set of Plutus Core builtins SHOULD be included in a CIP and should follow both the usual CIP process and the process laid out in this document.
+
+In order to move to Draft status, it MUST include:
+- In the Specification:
+    - Names and types/kinds for the new functions or types.
+    - A source for the implementation (e.g. a library which can be linked against); or a generic description of the functionality which is implementable in any programming language.
+    - In the Rationale:
+    - An argument for the utility of the new builtins.
+
+It SHOULD also include:
+- In the Rationale
+    - Examples of real-world use cases where the new additions would be useful.
+    - A comparison with an implementation using the existing features, and an argument why the builtin is preferable (e.g. better performance).
+
+In order to move to Proposed status, it MUST include:
+- In the Specification:
+    - For new types: a precise description of the measure used for the size of a value of that type.
+    - For new builtin functions: a costing function for the builtin function.
+- In the Rationale:
+    - If an external implementation is provided: an argument that it is trustworthy.
+    - Discussion of how any measures and costing functions were determined.
+- In the Path to Active section:
+    - A named implementor (person or organization) who will implement the support for the new builtins in the Plutus Core interpreter, or a pull request to the `plutus` repository with an implementation.
+
+In order to move to Active status, the following must be true:
+- The external implementations must be available.
+- The `plutus` repository MUST be updated with an implementation including costing.
+- The Plutus Core specification MUST be updated to include the new builtins.
+- The ledger MUST be updated to include new protocol parameters to control costing of the new builtins.
+- The completion of these steps MUST be tracked in the Path to Active section.
```

Comment:
All the issues raised in the 'move to Proposed' section apply here as well. In addition:

* No mention of testing. Given issues such as [this](https://github.com/input-output-hk/plutus/pull/4284#issuecomment-996185889), [this](https://github.com/input-output-hk/plutus/pull/4322#discussion_r782817268), and (while not from Plutus, _critically_ relevant) [this](https://github.com/input-output-hk/cardano-base/pull/253), the importance and quality of tests cannot be overstated. We also need to formalize practices such as use of Hedgehog, 1-in-a-thousand as the acceptable failure standard for Hedgehog's bad coverage implementation, and so forth: all the above are described in the links.

Reply:
As discussed in my main comment, this is an implementation issue so is out of scope.